config/routes.rb

```diff
@@ -0,0 +1,13 @@
+Rails.application.routes.draw do
+  root to: "works#top_ten"
+  get "/login", to: "users#login_form", as: :login
+  post "/login", to: "users#login"
+  post "/logout", to: "users#logout", as: :logout
+  patch "/works/:id/upvote", to: "votes#upvote", as: :upvote
+  resources :votes
```

Comment:
This creates the 7 restful routes for the VotesController even though none of those exist. Because the only action this controller has is the upvote action, this line is not needed.

Reply:
You can run `rails routes` from the terminal to see all the routes setup that are non-existent in the controller.